creating_reference_files/Generic_Instructions_for_Creating_ReferenceFiles.ipynb

```diff
@@ -0,0 +1,256 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "id": "f79e8d05",
+   "metadata": {},
+   "source": [
+    "Generic Instructions for Creating  New Reference Files\n",
+    "=======================================\n",
+    "\n",
+    "Tutorial Approach\n",
+    "---------------------\n",
+    "\n",
+    "We recommend that you review the tutorial for creating Flat Field Reference files \n",
+    "before doing this one.\n",
+    "\n",
+    "Rather than create a tutorial for each specific kind of reference file, this\n",
+    "tutorial will show how to use either the schema for that reference file as a \n",
+    "guide, or a simplified representation of the schema (the latter should be much\n",
+    "easier, and such information will be added to this tutorial as new reference \n",
+    "file types are added.\n",
+    "\n",
+    "This tutorial will use a few examples of how to do this,\n",
+    "and we expect that it will illustrate how to do the same for any reference file type.\n",
+    "More than one example is necessary since no particular one addresses all the \n",
+    "various aspects one may encounter.\n",
+    "\n",
+    "First, common elements to all reference files. All reference files have a meta\n",
+    "attribute, so one may do the following for all cases"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "b2890d2b",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import roman_datamodels.datamodels as rdd\n",
+    "import roman_datamodels.stnode as rds\n",
+    "import numpy as np\n",
+    "import asdf\n",
+    "from astropy.time import Time\n",
+    "# create meta dictionary and populate with common elements\n",
+    "meta = {'telescope': 'ROMAN', 'reftype': '<the expected value for this refernece file>'}\n",
+    "# Create the information about instrument configuration for this flat field.\n",
+    "# The first item is always the same, but the others depend on the particular flat field\n",
+    "instrument = {'name': 'WFI', 'detector': 'WFI01', 'optical_element': 'F129'}\n",
```

Comment:
How will we handle OR'ed CRDS match patterns?  For example a detector value of "WFI01|WFI02|WFI03...".  Will Roman need to support those?

Reply:
I'm not sure I understand. The reference file will need an OR'ed specification? The match is mainly for the data file itself. So I guess you are asking how this intent should be represented in the value for that field in the reference file. That seems tricky since it really is only for documentation and no specific driver for the matching criteria (or I may have forgotten that there is a way of extracting the match from reference file on submission). Nevertheless, one can change matching criteria after a file is submitted and the intent of its usage may change without requiring resubmission.

We could just loosen the schema to allow a pattern or something. Would that be ok?

Reply:
The CRDS website does extract match criteria from the file and use it to automatically update the .rmap.  JWST handles this by defining additional metadata fields that allow patterns, for example `meta.instrument.p_detector`:

https://github.com/spacetelescope/jwst/blob/master/jwst/datamodels/schemas/keyword_pdetector.schema.yaml

Reply:
Does that match any OR combination?

Reply:
It does, yeah.  The first part of the pattern has regex ORs that select a single detector:

NRCA1|NRCA2|NRCA3...

then there's a literal `|` surrounded by optional whitespace:

\\s*\\|\\s*

and that whole thing is repeated 1-or-more times.

Reply:
Sounds like we need to add the `p_` attributes to the schemas

Reply:
Apparently we don't according to Lisa.